client/client.go

```diff
@@ -259,8 +260,30 @@ func (c *Cmd) Dial() error {
 	if err := c.UserKeyConfig(); err != nil {
 		return err
 	}
-	addr := net.JoinHostPort(c.HostName, c.Port)
-	cl, err := ssh.Dial(c.network, addr, &c.config)
+	// Sadly, no vsock in net package.
+	var (
+		cl   *ssh.Client
+		err  error
+		addr string
+	)
+
+	switch c.network {
+	case "vsock":
+		id, port, err := vsockIdPort(c.HostName, c.Port)
```

Comment:
do you need to check err on this line?

Reply:
yeah, thanks, and I added a test for vsockIdPort